### ISSUE TITLE
Stderr and non-zero exit status aren't used in response to a malformed comment directive

### ISSUE THREAD

**Issue by frou** [kkinnear/zprint#160]
Hi there. Given this input (with the malformed  `;!zprint` directive)

```
#!/usr/bin/env bb

;!zprint {:format

(ns hello
  (:require [clojure.java.io :refer [file]]
            [clojure.java.shell :refer [sh]]))
```

The output is this:

```
Warning:  Unable to create zprint options-map from: '{:format
' found in !zprint directive number: 1 because: java.lang.RuntimeException: EOF while reading
#!/usr/bin/env bb

;!zprint {:format

(ns hello
  (:require [clojure.java.io :refer [file]]
            [clojure.java.shell :refer [sh]]))
```

The warning text is mixed in with the code on stdout, and zprint's exit status is zero (meaning that as far as my editor's format-on-save was concerned, it was a complete success, and so that warning text got inserted into the source code).

I searched the issue tracker and this seems pretty similar to [an old resolved issue](https://github.com/kkinnear/zprint/issues/69), just in a different context. Maybe zprint's exit status should remain zero (since there's just a warning, not an error) but it seems like the warning text should definitely go to stderr.

---

zprint 1.0.0, macOS

**Comment by kkinnear** [comment]
Thanks for catching this -- and for a very clear example and description of what you expected would happen!  

In the next release (`1.0.2`, coming soon), it will output the unformatted input to stdout, the error text to stderr, and exit with an exit status of 1.  Which is what I think you expected.  The underlying code to do the errors to stderr and unformatted input to stdout and exit status of 1 was all there, but the routine that detected the comment API failure (the `;!zprint {:format` error), needed to rethrow the Exception -- which it now does.   Hopefully that will now make this work as you (and I) expected.

**Comment by frou** [comment]
@kkinnear Cool, I appreciate it!

**Comment by kkinnear** [comment]
Fixed in 1.0.2.  Thanks again for noticing this!
